File: Problem 15/ans.cpp
#include<iostream>

using namespace std;

int main()
{
    long long int a[21][21];
    for(int b=0;b<21;b++)
    {
        a[b][0]=1;
        a[0][b]=1;
    }
    for(int b=1;b<21;b++)
        for(int c=1;c<21;c++)
            a[b][c]=a[b-1][c]+a[b][c-1];
    cout << "Number of routes: " << a[20][20] << endl;
    return 0;
}

File: Problem 15/ans.py
import numpy as np

a=np.zeros([21,21],dtype='int64')
a[:,0]=1
a[0,:]=1
for i in range(1,a.shape[0]):
    for j in range(1,a.shape[1]):
        a[i,j]=a[i-1,j]+a[i,j-1]
print("Number of routes:",a[a.shape[0]-1,a.shape[1]-1])